src/main/java/org/andersen/homework/model/entity/Ticket.java

```diff
@@ -4,40 +4,118 @@
-import lombok.Builder;
 import lombok.Getter;
-import lombok.NoArgsConstructor;
 import lombok.Setter;
-import lombok.ToString;
+import org.andersen.homework.annotation.NullableWarning;
 import org.andersen.homework.model.enums.StadiumSector;
 
 @Getter
 @Setter
-@ToString
-@Builder
-@NoArgsConstructor
 @AllArgsConstructor
 public class Ticket {
 
   @Max(999)
   @Min(0)
-  private Short id;
+  @NullableWarning
+  private final Short id;
 
-  private Float priceInUsd;
+  @NullableWarning
+  private final Float priceInUsd;
 
   @Size(max = 10)
-  private String concertHallName;
+  @NullableWarning
+  private final String concertHallName;
 
   @Max(999)
   @Min(0)
-  private Short eventCode;
+  @NullableWarning
+  private final Short eventCode;
 
-  private Boolean isPromo;
+  @NullableWarning
+  private final Boolean isPromo;
 
+  @NullableWarning
+  private final Float backpackWeightInKg;
+
+  @NullableWarning
   private LocalDateTime time;
 
+  @NullableWarning
   private StadiumSector stadiumSector;
 
-  private Float backpackWeightInKg;
+  public Ticket(Short id, Float priceInUsd, String concertHallName, Short eventCode,
+      Boolean isPromo,
+      Float backpackWeightInKg) {
+    this.id = id;
+    this.priceInUsd = priceInUsd;
+    this.concertHallName = concertHallName;
+    this.eventCode = eventCode;
+    this.isPromo = isPromo;
+    this.backpackWeightInKg = backpackWeightInKg;
+    checkFields();
+  }
+
+  public void share(String phoneNumber) {
+    System.out.println("Ticket was shared by phone: " + phoneNumber + ".");
+  }
+
+  public void share(String phoneNumber, String email) {
+    System.out.println("Ticket was shared by phone: " + phoneNumber + " and email: " + email + ".");
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Ticket ticket = (Ticket) o;
+    return Objects.equals(id, ticket.id) && Objects.equals(priceInUsd,
+        ticket.priceInUsd) && Objects.equals(concertHallName, ticket.concertHallName)
+        && Objects.equals(eventCode, ticket.eventCode) && Objects.equals(isPromo,
+        ticket.isPromo) && Objects.equals(backpackWeightInKg, ticket.backpackWeightInKg)
+        && Objects.equals(time, ticket.time) && stadiumSector == ticket.stadiumSector;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(id, priceInUsd, concertHallName, eventCode, isPromo, backpackWeightInKg,
+        time,
+        stadiumSector);
+  }
+
+  @Override
+  public String toString() {
+    return "Ticket{" +
+        "id=" + id +
+        ", priceInUsd=" + priceInUsd +
+        ", concertHallName='" + concertHallName + '\'' +
+        ", eventCode=" + eventCode +
+        ", isPromo=" + isPromo +
+        ", backpackWeightInKg=" + backpackWeightInKg +
+        ", time=" + time +
+        ", stadiumSector=" + stadiumSector +
+        '}';
+  }
+
+  private void checkFields() {
```

Comment:
Does this method connected with Ticket itself? Is it an ability of a ticket? :)
What if we would like to use the annotation for other class?